src/resolve/resolved-dns-dnssec.c

```diff
@@ -613,6 +707,9 @@ int dnssec_verify_rrset(
         gcry_md_hd_t md = NULL;
         int r, md_algorithm;
         size_t k, n = 0;
+        FILE *f = NULL;
+        size_t sig_size = 0;
```

Comment:
Please use _cleanup_free_ and _cleanup_fclose_ on these and avoid manual freeing/closing